models/Document/Editable/Areablock.php

```diff
@@ -248,7 +248,11 @@ protected function getEditableHandler(): EditableHandler
 
     public function setDataFromResource(mixed $data): static
     {
-        $unserializedData = Tool\Serialize::unserialize($data);
+        if (is_string($data)) {
+            $unserializedData = Tool\Serialize::unserialize($data);
+        } else {
```

Comment:
We could remove the else and just set $unserializedData = $data as default no?

Reply:
@mattamon This was possible, but I make it more type save. Now it was possible to set a wrong type and nothing happens. I think it should be throw a InvalidArgumentException() like in the renderlet editable.